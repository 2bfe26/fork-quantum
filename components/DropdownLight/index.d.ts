
// Generated with scripts/create-component.js
import { Component } from 'react';

export interface DropdownLightProps {
  someProp?: string;
}

export default class DropdownLight extends Component<DropdownLightProps> {}
    