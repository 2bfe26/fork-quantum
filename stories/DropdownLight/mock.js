export const itemsObjectMock = [
  { value: 1, label: 'Lemon' },
  { value: 2, label: 'Banana' },
  { value: 3, label: 'Strawberry' },
  { value: 4, label: 'Orange' },
  { value: 5, label: 'Avocado' },
];

export const itemsStringMock = [
  'Lemon',
  'Banana',
  'Strawberry',
  'Orange',
  'Avocado',
];
