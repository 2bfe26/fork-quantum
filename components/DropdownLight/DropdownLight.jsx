import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colors, spacing, baseFontSize } from '../shared/theme';
import { itemsMock } from '../../stories/DropdownLight/mock';

const itemPropType = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.shape({
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    label: PropTypes.string,
  }),
]);

const DropList = styled.select`
  width: 100%;
  height: 44px;
  font-size: 16px;
`;
const OptionsItems = styled.option`
  font-size: 16px;
`;

const DropdownLight = ({
  disabled,
  label,
  placeholder,
  items,
  selectedItem,
  onChange,
  theme,
}) => (
  <>
    <DropList
      disabled={disabled}
      onChange={onChange}
      label={label}
      placeholder={placeholder}
      items={items}
      selectedItem={selectedItem}
      theme={theme}
    >
      {itemsMock.map(item => (
        <OptionsItems key={itemsMock}>{item}</OptionsItems>
      ))}
    </DropList>
  </>
);

DropdownLight.propTypes = {
  disabled: PropTypes.bool,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  items: PropTypes.arrayOf(itemPropType),
  selectedItem: itemPropType,
  onChange: PropTypes.func,
  theme: PropTypes.shape({
    colors: PropTypes.object,
    spacing: PropTypes.object,
    baseFontSize: PropTypes.number,
  }),
};

DropdownLight.defaultProps = {
  disabled: false,
  label: '',
  placeholder: 'Select an option',
  selectedItem: null,
  items: [],
  onChange: () => {},
  theme: { colors, spacing, baseFontSize },
};

export default DropdownLight;
