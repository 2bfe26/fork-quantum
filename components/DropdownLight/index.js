// Generated with scripts/create-component.js
import DropdownLight from './DropdownLight';

export default DropdownLight;
