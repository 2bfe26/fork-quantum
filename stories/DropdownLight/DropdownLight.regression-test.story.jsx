// Generated with scripts/create-componnent.js

import DropdownLight from '../../components/DropdownLight';

export default {
  title: 'DropdownLight',
  component: DropdownLight,
};

const Template = args => <DropdownLight {...args} />;

export const Default = Template.bind({});

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};
