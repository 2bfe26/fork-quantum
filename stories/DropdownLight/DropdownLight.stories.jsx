import DropdownLight from '../../components/DropdownLight/DropdownLight';
import { itemsMock } from './mock';

const Template = args => {
  const { items = itemsMock, ...rest } = args;
  return <DropdownLight {...rest} items={items} />;
};

export const Controlled = () => {
  const [fruits] = useState([
    'Lemon',
    'Banana',
    'Strawberry',
    'Orange',
    'Avocado',
  ]);
  const [selectedFruit, setSelectedFruit] = useState('');

  return (
    <DropdownLight
      label="Fruits"
      placeholder="Select a fruit"
      items={fruits}
      selectedItem={selectedFruit}
      onChange={item => setSelectedFruit(item)}
    />
  );
};

export const Default = Template.bind({});
