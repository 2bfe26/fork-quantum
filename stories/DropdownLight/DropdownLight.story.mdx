<!-- Generated with scripts/create-componnent.js -->
import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import  DropdownLight  from '../../components/DropdownLight';

import { Header } from '../shared';

import { 
  Default,
  Disabled,
  WithArrayOfString,
  SelectedItemControlled
} from './DropdownLight.stories.jsx';

<Meta title="DropdownLight" component={DropdownLight}  />

<Header title="DropdownLight">
      A dropdown receives user input, allowing it to choose one of several options.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)


## <a id="importing"></a>Importing and usage

```js
import { DropdownLight } from '@catho/quantum';
```

<Canvas>
  <Story story={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={DropdownLight} />

## <a id="examples"></a>Examples

### Disabled

<Canvas>
  <Story story={Disabled}/>
</Canvas>

### WithArrayOfString

<Canvas>
  <Story story={WithArrayOfString}/>
</Canvas>

### With a item selected

<Canvas>
  <Story story={SelectedItemControlled}/>
</Canvas>
