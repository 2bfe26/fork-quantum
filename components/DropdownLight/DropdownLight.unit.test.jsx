// Generated with scripts/create-component.js
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import DropdownLight from './DropdownLight';

import {
  itemsStringMock,
  itemsObjectMock,
} from '../../stories/DropdownLight/mock';

describe('<DropdownLight />', () => {
  it('should match the snapshots', () => {
    expect(
      render(<DropdownLight items={itemsStringMock} />).asFragment(),
    ).toMatchSnapshot();
  });

  it('should return a value from item, using a object items', () => {
    render(<DropdownLight items={itemsObjectMock} />);

    const dropdown = screen.getByRole('button');
    userEvent.click(dropdown);

    const optionItem = screen.getByText('Lemon');
    userEvent.click(optionItem);

    const input = Number(screen.getByLabelText('selecione uma opcao').value);

    const lemonItem = itemsObjectMock[0].value;

    expect(input).toEqual(lemonItem);
  });

  it('should return a value from item, using a string items', () => {
    render(<DropdownLight items={itemsStringMock} />);

    const dropdown = screen.getByRole('button');
    userEvent.click(dropdown);

    const optionItem = screen.getByText('Lemon');
    userEvent.click(optionItem);

    const input = screen.getByLabelText('selecione uma opcao').value;

    const lemonItem = itemsStringMock[0];

    expect(input).toEqual(lemonItem);
  });
});
