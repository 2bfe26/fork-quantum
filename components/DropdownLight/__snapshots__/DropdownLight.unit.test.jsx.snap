// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownLight /> should match the snapshots 1`] = `
<DocumentFragment>
  <input
    aria-label="selecione uma opcao"
    type="hidden"
    value=""
  />
  .c1 {
  display: inline-block;
  pointer-events: none;
  width: 24px;
}

.c0 {
  width: 100%;
  padding: 8px 16px;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: left;
  margin-top: 0px;
  -webkit-transition: all 0.2s ease-in-out;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0:hover {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgb(18 80 196 / 20%),0px 2px 2px 0px rgb(18 80 196 / 14%),0px 1px 5px 0px rgb(18 80 196 / 12%);
}

.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgb(18 80 196 / 20%),0px 2px 2px 0px rgb(18 80 196 / 14%),0px 1px 5px 0px rgb(18 80 196 / 12%);
}

<button
    class="c0"
  >
    Select an option
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root c1"
      data-qtm-preloader="icon"
      focusable="false"
      style="font-size: 24px; max-width: 24px;"
      theme="[object Object]"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
